.genai/io/capstone/rag/stdlib_docs.py

```
@@ -0,0 +1,459 @@
+            raise ValueError("embeddings_client required to generate embeddings")
+
+        # Generate embeddings for all docs
+        self._doc_texts = [self._doc_to_text(doc) for doc in self.docs]
+        self._doc_embeddings = self.embeddings_client.embed_batch(self._doc_texts)
+
+        # Cache for next time
+        self._save_cache()
+
+    def get_doc_by_module(self, module_name: str) -> List[StdlibDoc]:
+        """Get all docs for a specific module."""
+        return [doc for doc in self.docs if doc.module == module_name]
+
+    def get_doc_by_name(self, name: str) -> Optional[StdlibDoc]:
+        """Get doc for a specific function/class name."""
+        for doc in self.docs:
+            if doc.name == name:
+                return doc
+        return None
+
+    def search_relevant_docs(
+        self,
+        query: str,
+        top_k: int = 3,
+        module_filter: List[str] = None
+    ) -> List[Tuple[StdlibDoc, float]]:
+        """
+        Search for relevant stdlib docs using semantic similarity.
+
+        Args:
+            query: Search query (code snippet or description)
+            top_k: Number of results to return
+            module_filter: If provided, only search these modules
+
+        Returns:
+            List of (StdlibDoc, score) tuples, sorted by relevance
+        """
+        if self._doc_embeddings is None:
+            self.initialize()
+
+        if self.embeddings_client is None:
+            # Fallback to keyword matching if no embeddings client
+            return self._keyword_search(query, top_k, module_filter)
+
+        # Get query embedding
+        query_embedding = self.embeddings_client.embed_text(query)
+
+        # Calculate cosine similarity with all docs
+        results = []
+        for i, (doc, doc_embedding) in enumerate(zip(self.docs, self._doc_embeddings)):
+            # Apply module filter
+            if module_filter and doc.module not in module_filter:
+                continue
```

392 : The broad `except Exception:` with a `pass` in `_load_cache` silently swallows all errors. If a persistent issue like a file permission error or data corruption occurs, the cache will fail to load and be regenerated on every run, degrading performance without any warning. The exception should be logged to make such problems visible for debugging.